content/blog/larkfid-twoja-baza-wiedzy/index.md

```diff
@@ -0,0 +1,115 @@
+
+#### Ekstrakcja informacji (faktów)
+
+Każdy artykuł/post/wpis to tylko tekst ze zdjęciami, nie zawiera on metadanych o miejscu/wydarzeniu, które można by odczytać. Aby to zrobić, trzeba przetworzyć pobrany tekst w poszukiwaniu dat, nazw miejsc, osób i nazwisk. Do tego zadania można wykorzystać np. Morfeusza 2 (dla języka polskiego), modele LLM (ChatGPT, LLama), i techniki NLP: Entity recognition.
+
+#### Deduplikacja
+
+Deduplikacja to jest jeden z trudniejszych problemów, bo dla każdego z artykułów należy zdecydować, czy artykuł o tym samym temacie/treści nie znajduje się już w bazie. Żeby to zrobić, można porównać:
+
+1. Tytuły — czy są identyczne/przekazują tę samą informację różnymi słowami?
+2. Treści — identycznie jak Tytuły.
+3. Datę publikacji — bliska data publikacji sugeruje pewien związek (być może powielenie!)
+4. Lokalizację wydarzenia (jeżeli występuje)
+5. Inne powiązania, wynikające np. z bliskości źródeł np. strona www i fb tej samej firmy.
+
+Technologie: Python (NLP: Text similarity, Semantic Textual Similarity), Mozilla Readablility, Morfeusz 2.
+
+## Implementacja
+
+_Nie będę ukrywać, że po pierwsze nie mieliśmy doświadczenia w tworzeniu tego typu aplikacji, a po drugie: zbyt wcześnie myśleliśmy o skali, o tym, jak wiele wiadomości możemy przetworzyć i jak zaprojektować system tak by najpierw działał. Więc to, co stworzyliśmy, było zdecydowanie zbyt skomplikowane. Natomiast pozwoliło nam "pobawić" się wieloma ciekawymi technologiami i napotkać problemy, których inaczej napotkać się nie da. Była to dla nas wszystkich cenna lekcja._
+
+Oto cała architektura:
+<image src="larkfid-arch.png" alt="Architektura aplikacji LARKFid" />
+
+> Btw. to, że architekturę aplikacji tworzą małe serwisy (które można by nazwać mikro), wcale nie oznacza, że architekturą tej aplikacji są mikroserwisy. Wręcz przeciwnie, architektura LARKFid to monolit, gdyż zależności między kolejnymi serwisami są bardzo duże (wszystkie serwisy tworzą jeden potok danych).
+
+Jej podstawą jest message broker — RabbitMQ. Jego zadaniem jest przekazywanie wiadomości między serwisami. Pierwszą częścią architektury jest "scheduler" i "rss-downloader", odpowiadają one za pobieranie danych z feedów RSS i indeksowanie linków do artykułów w nich zawartych.
+
+Następnym krokiem jest wydanie zadania do zindeksowania danego artykułu (z wcześniej zindeksowanego linku), zadanie to wykonuje "Article downloader". Po pobraniu artykułu dodatkowo wydobywa on z niego treść przy wykorzystaniu Readability od Mozilli.
+
+Ostatnim krokiem jest utworzenie wektorów z wydobytych tekstów (embeddings), które posłużą później do wykonywania zapytań przy wykorzystaniu języka naturalnego. Silnikiem tych zapytań jest OpenSearch KNN Plugin. (Artykuł na ten temat stworzyłem tu: [How to use OpenSearch k-NN as a semantic search engine ](https://dev.to/finloop/how-to-use-opensearch-k-nn-as-a-semantic-search-engine-je9)). Oto przykładowa interakcja z wyszukiwaniem z wykorzystaniem kNN:
```

Comment:
Link do artykułu to powinien być przypis

Reply:
Footnotes są zbugowane: https://github.com/getzola/zola/issues/1070

muszę dodać kawałek JSa, który to naprawi

Reply:
A jednak nie, kawałek css'a wystarszył. Ale zajęło mi to zdecydowanie zbyt dużo czasu :disappointed: